sample-app/src/components/SearchBar.tsx

```diff
@@ -1,24 +1,59 @@
-import { useAnswersActions, useAnswersState } from '@yext/answers-headless-react';
+import { AutocompleteResult, useAnswersActions, useAnswersState } from '@yext/answers-headless-react';
 import InputDropdown from './InputDropdown';
-import renderWithHighlighting from './utils/renderWithHighlighting';
+import renderHighlightedValue from './utils/renderHighlightedValue';
 import { ReactComponent as MagnifyingGlassIcon } from '../icons/magnifying_glass.svg';
-import '../sass/SearchBar.scss';
-import '../sass/Autocomplete.scss';
+import { ReactComponent as YextLogoIcon } from '../icons/yext_logo.svg';
 import LoadingIndicator from './LoadingIndicator';
 import { useAutocomplete } from '../hooks/useAutocomplete';
 import DropdownSection from './DropdownSection';
 import { processTranslation } from './utils/processTranslation';
 import { useRef } from 'react';
 import { AutocompleteResponse, SearchIntent } from '@yext/answers-headless-react';
 import { executeSearch, updateLocationIfNeeded } from '../utils/search-operations';
+import { useComposedCssClasses, CompositionMethod } from '../hooks/useComposedCssClasses';
 
 const SCREENREADER_INSTRUCTIONS = 'When autocomplete results are available, use up and down arrows to review and enter to select.'
 
+const builtInCssClasses: SearchBarCssClasses = {
+  container: 'h-12',
+  divider: 'border mx-2',
+  dropdownContainer: 'relative bg-white py-1 z-10',
+  focusedOption: 'bg-gray-100',
+  inputContainer: 'h-12 inline-flex items-center justify-between w-full',
+  inputDropdownContainer: 'bg-white shadow border rounded-3xl border-gray-300 w-full overflow-hidden',
+  inputElement: 'outline-none flex-grow border-none h-full px-2',
+  logoContainer: 'w-8 mx-2',
+  option: 'flex items-center py-1 px-2 cursor-pointer',
+  resultIconContainer: 'opacity-20 w-8 h-8 pl-1 mr-4',
+  searchButtonContainer: 'w-8 h-full mx-2',
+  submitButton: 'h-full w-full'
+}
+
+interface SearchBarCssClasses {
```

Comment:
Instead of defining SearchBarCssClasses here, I wonder if we could do something like 
```ts
interface SearchBarCssClasses extends DropdownSectionCssClasses, InputDropdownCssClasses {
//extra classes
}
``` 
Unfortunately I don't think this addresses conflicting names. I had a thought of breaking up SearchBarCssClasses into smaller sub objects i.e
```ts
interface SearchBarCssClasses {
  DropdownSection: DropdownSectionCssClasses,
  InputDropdownCssClasses: InputDropdownCssClasses
}
```
But I don't really like how this bleeds implementation details. Instead of saying specifically
`DropdownSection: DropdownSectionCssClasses`. We could be more general about how we name the sub objects to not bleed details but even then I don't really like the extra complexity. I don't really have a solution just throwing up ideas.

Reply:
I think the extending makes sense just to clean things up even if it doesn't address the naming conflicting issue

Reply:
I updated to extend the DropdownSectionCssClasses and the InputDropdownCssClasses